src/patterns/security-code/index.md.njk

```diff
@@ -0,0 +1,111 @@
+
+## When to use this pattern
+
+Use this pattern if your users need to sign in to your service.
+
+Requiring users to enter a security code sent to their mobile phone provides a second layer of security over a [password](/patterns/passwords/).
+
+You can require a security code on every sign in or only once per device, depending upon the risk of your service.
+
+## How it works
+
+You should first ask users for their [password](/patterns/passwords/). Once they have successfully entered this, send a text message to their mobile phone with a 5 digit code in this format:
+
+{% from "govuk/components/inset-text/macro.njk" import govukInsetText %}
+{{ govukInsetText({
+  text: "12345 is your [service name] security code"
+}) }}
+
+Then ask the user to enter this code:
+
+{{ example({group: "patterns", item: "security-code", example: "default", html: true, nunjucks: true, open: true}) }}
+
+### Missing phone
+
+You should advise users what to do if they no longer have access to their phone.
+
+Depending upon your service, they may need to contact a support address, or may be able to ask a colleague to update their phone number.
+
+### Expiry
+
+There should be a time limit of 15 minutes for users to enter the code. If this time limit expires, display an error and send the user a new code:
+
+{{ example({group: "patterns", item: "security-code", example: "error-expired", html: true, nunjucks: true, open: false}) }}
+
+### Resending
+
+You should also allow users to request a new code, if the first one does not arrive, as text messages can be delayed.
+
+Use this page, via the "Not received a text message?" link:
+
+{{ example({group: "patterns", item: "security-code", example: "resend", html: true, nunjucks: true, open: false}) }}
+
+This should send a new code, with its own 15 minute expiry window. However the previous code should remain valid within its own expiry time.
+
+### First time usage
+
+When the user first enters their mobile number, for example when setting up an account for your service, verify their mobile number straight away by sending a security code and asking for it.
+
+If the user follows the ‘Not received a text message?’ link, allow them to check which mobile number they entered, and to change it if necessary. This prevents the user becoming stuck if they entered a mobile number incorrectly.
+
+{{ example({group: "patterns", item: "security-code", example: "resend-first-time", html: true, nunjucks: true, open: false}) }}
+
+### Error messages
+
+Error messages should be styled like this:
+
+{{ example({group: "patterns", item: "security-code", example: "error-incorrect", html: true, nunjucks: true, open: false}) }}
+
+If the user doesn’t enter enough digits:
+
+Say ‘You haven’t entered enough numbers’.
+
+If the user enters too many digits:
+
+Say ‘You’ve entered too many numbers’.
+
+If the user enters non-numeric characters, other than spaces:
```

Comment:
This should be more explicit. It’s important to treat `123 45` as valid and equivalent to `12345`. When I tried dictating a code with Dragon Naturally Speaking I got the former not the latter.

Reply:
@quis any suggestions on how to make this more explicit? Perhaps we should add something to the "How it works" section about removing all whitespace characters when checking the code?

Reply:
The [payment card details](https://design-system.service.gov.uk/patterns/payment-card-details/) pattern says:
> # How it works
>
> ## Allow different formats
>
> Let users enter payment card numbers in whatever format is familiar to them. Allow additional spaces, hyphens and dashes.

I think a similar line after ‘Then ask the user to enter this code:’ would be good in this documentation.

Reply:
Resolve, edit has added in line 46